isthmus/src/main/java/io/substrait/isthmus/expression/RexExpressionConverter.java

```diff
@@ -73,7 +92,115 @@ public Expression visitLiteral(RexLiteral literal) {
+          builder.aggregateFunction(aggMeasure).hasNormalAggregateFunction(true);
+        } else if (functionInvocation instanceof WindowFunctionInvocation) {
+          var windowFuncInvocation = (WindowFunctionInvocation) functionInvocation;
+          var windowFunc =
+              ImmutableExpression.WindowFunction.builder().function(windowFuncInvocation).build();
+          builder.windowFunction(windowFunc).hasNormalAggregateFunction(false);
+        } else {
+          throw new RuntimeException(
+              String.format("Unexpected AbstractFunctionInvocation :%s", functionInvocation));
+        }
+      }
+    }
+    if (functionInvocation == null) {
+      throw new RuntimeException(
+          String.format(
+              "Not found the corresponding window aggregate function:%s", sqlAggFunction));
+    }
+    var window = over.getWindow();
+    var partitionExps = window.partitionKeys.stream().map(r -> r.accept(this)).toList();
+    var sortFields = window.orderKeys.stream().map(r -> toSortField(r)).toList();
+
+    return builder
+        .addAllOrderBy(sortFields)
+        .addAllPartitionBy(partitionExps)
+        .lowerBound(lowerBound)
+        .upperBound(upperBound)
+        .type(TypeConverter.convert(over.getType()))
+        .build();
+  }
+
+  private WindowBound toWindowBound(RexWindowBound rexWindowBound) {
+    if (rexWindowBound.isCurrentRow()) {
+      return ImmutableWindowBound.CurrentRowWindowBound.builder().build();
+    }
+    if (rexWindowBound.isUnbounded()) {
+      var direction =
+          rexWindowBound.isFollowing()
+              ? WindowBound.Direction.FOLLOWING
+              : WindowBound.Direction.PRECEDING;
+      return ImmutableWindowBound.UnboundedWindowBound.builder().direction(direction).build();
+    } else {
+      var direction =
+          rexWindowBound.isFollowing()
+              ? WindowBound.Direction.FOLLOWING
+              : WindowBound.Direction.PRECEDING;
+      var offset = rexWindowBound.getOffset().accept(this);
+      return ImmutableWindowBound.BoundedWindowBound.builder()
+          .direction(direction)
+          .offset(offset)
+          .build();
+    }
+  }
+
+  private Expression.SortField toSortField(RexFieldCollation rexFieldCollation) {
+    var expr = rexFieldCollation.left.accept(this);
+    var rexDirection = rexFieldCollation.getDirection();
+    Expression.SortDirection direction =
+        switch (rexDirection) {
+          case ASCENDING -> rexFieldCollation.getNullDirection()
+                  == RelFieldCollation.NullDirection.LAST
+              ? Expression.SortDirection.ASC_NULLS_LAST
+              : Expression.SortDirection.ASC_NULLS_FIRST;
+          case DESCENDING -> rexFieldCollation.getNullDirection()
+                  == RelFieldCollation.NullDirection.LAST
+              ? Expression.SortDirection.DESC_NULLS_LAST
+              : Expression.SortDirection.DESC_NULLS_FIRST;
+          default -> throw new RuntimeException(
```

Comment:
Is it better to use `IllegalArgumentException` here?

Reply:
done